src/components/MapContainer/Layers/googleSheets.tsx

```diff
@@ -0,0 +1,79 @@
+import React, { useContext, useEffect } from 'react'
```

Comment:
Nice layer! --> There is a refactor in progress regarding the original VectorLayer.tsx and I have a bit of design pattern intent that might give more direction.  🐋

In the end, we likely want to reuse VectorLayer as a reusable component by passing it specific props related to each data source (style, source information, etc).  This way, instead of copying large, slightly modified blocks of code that are pretty similar, we can just pass the info needed for each instance of a VectorLayer as props from above.

Reply:
Thanks

Reply:
will sure look into the upcoming changes as well

Reply:
@ivanoats could u review merge this one if its suitable?
so that further changes on this PR or new PRs can be made on the new points

Reply:
@epachols what was the status of the code from the hackathon as it relates to this PR? Should we go with @Devesh21700Kumar 's changes here or is there other code to consider?